core/domain/stats_services.py

```diff
@@ -1124,6 +1124,87 @@ def get_learner_answer_details(entity_type, state_reference):
     return None
 
 
+def get_learner_answer_info_for_exploration(exp_id):
+    """Returns the learner answer info in dict format.
+
+    Args:
+        exp_id: str. The ID of the exploration.
+
+    Returns:
+        list. Consists of dicts
+            with state name as keys, and a list of learner answer info dicts as
+            values. The learner answer info dicts contains following keys:
+                learner_answer_info_id: str. The id of the LearnerAnswerInfo
+                    object.
+                answer: json-like. The answer which is submitted by the
+                    learner. The actual type of answer depends on the
+                    interaction.
+                answer_details: str. The details the learner will submit when
+                    the learner will be asked questions like 'Hey how did you
+                    land on this answer', 'Why did you pick that answer' etc.
+                    i.e the learner's explanation of their choice.
+                created_on: datetime. The time at which the answer details
+                    was received.
+    """
+    exp = exp_fetchers.get_exploration_by_id(exp_id)
+    learner_answer_info_data = []
+    for state_name in exp.states.keys():
+        state_reference = get_state_reference_for_exploration(
+            exp_id, state_name)
+        learner_answer_details = get_learner_answer_details(
+            feconf.ENTITY_TYPE_EXPLORATION, state_reference)
+        if learner_answer_details is not None:
+            learner_answer_info_data.append({
+                'state_name': state_name,
+                'interaction_id': learner_answer_details.interaction_id,
+                'customization_args': exp.states[state_name].interaction
+                                      .to_dict()['customization_args'],
+                'learner_answer_info_dict_list': [
+                    learner_answer_info.to_dict() for learner_answer_info
+                    in learner_answer_details.learner_answer_info_list]
+            })
+    return learner_answer_info_data
+
+
+def get_learner_answer_info_for_question(question_id):
+    """Returns the learner answer info in dict format.
+
+    Args:
+        question_id: str. The ID of the question.
+
+    Returns:
+        list. Consists of learner answer info
```

Comment:
Ditto.

Reply:
Made services to return domain object only